dspace-api/src/main/java/org/dspace/external/provider/impl/LiveImportDataProvider.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    /**
+     * This method set the SourceIdentifier for the ExternalDataProvider
+     * @param sourceIdentifier   The UNIQUE sourceIdentifier to be set on any LiveImport data provider
+     */
+    public void setSourceIdentifier(String sourceIdentifier) {
+        this.sourceIdentifier = sourceIdentifier;
+    }
+
+    /**
+     * This method set the MetadataSource for the ExternalDataProvider
+     * @param metadataSource {@link org.dspace.importer.external.service.components.MetadataSource} implementation used to process the input data
+     */
+    public void setMetadataSource(QuerySource querySource) {
+        this.querySource = querySource;
+    }
+
+    /**
+     * This method set dublin core identifier to use as metadata id
+     * @param recordIdMetadata dublin core identifier to use as metadata id
+     */
+    public void setRecordIdMetadata(String recordIdMetadata) {
+        this.recordIdMetadata = recordIdMetadata;
+    }
+
+    /**
+     * This method set the dublin core identifier to display the title
+     * @param displayMetadata metadata to use as title
+     */
+    public void setDisplayMetadata(String displayMetadata) {
+        this.displayMetadata = displayMetadata;
+    }
+
+    @Override
+    public Optional<ExternalDataObject> getExternalDataObject(String id) {
+        try {
+            ExternalDataObject externalDataObject = getExternalDataObject(querySource.getRecord(id));
+            return Optional.of(externalDataObject);
+        } catch (MetadataSourceException e) {
+            throw new RuntimeException(
+                    "The live import provider " + querySource.getImportSource() + " throws an exception", e);
+        }
+    }
+
+    @Override
+    public List<ExternalDataObject> searchExternalDataObjects(String query, int start, int limit) {
+        Collection<ImportRecord> records;
+        try {
+            records = querySource.getRecords(query, start, limit);
+            return records.stream().map(r -> getExternalDataObject(r)).collect(Collectors.toList());
+        } catch (MetadataSourceException e) {
+            throw new RuntimeException(
+                    "The live import provider " + querySource.getImportSource() + " throws an exception", e);
+        }
+    }
+
+    @Override
+    public boolean supports(String source) {
+        return StringUtils.equalsIgnoreCase(sourceIdentifier, source);
+    }
+
+    @Override
+    public int getNumberOfResults(String query) {
+        try {
+            return querySource.getRecordsCount(query);
+        } catch (MetadataSourceException e) {
+            throw new RuntimeException(
+                    "The live import provider " + querySource.getImportSource() + " throws an exception", e);
+        }
+    }
+
+    /**
+     * Internal method to convert an ImportRecord to an ExternalDataObject
+     * 
+     * FIXME it would be useful to remove ImportRecord at all in favor of the
+     * ExternalDataObject
```

Comment:
The fixme is mean to note that it would be useful to refactor the LiveImport framework to use directly the ExternalDataObject instead than it owns equivalent concept. This is out of scope of the current PR as it will imply a refactoring of the Live Import framework but essentially we found that the ExternalAuthority and the LiveImport have equivalent concepts that could be merged